chronos/transports/datagram.nim

```diff
@@ -1002,7 +1002,7 @@ proc getMessage*(transp: DatagramTransport): seq[byte] {.
     copyMem(addr res[0], addr transp.buffer[0], transp.buflen)
     res
   else:
-    default
+    @[]
```

Comment:
why allocation?

Reply:
there is no allocation for an empty seqs:  https://github.com/nim-lang/Nim/blob/v1.6.14/compiler/ccgexprs.nim#L1416C10-L1416C10

Reply:
I'm not very familiar with C compiler source code but i'm not sure that code you point to will be executed when compiler will encounter `@` symbol.
Because in `system.nim` there present
```nim
proc `@`* [IDX, T](a: sink array[IDX, T]): seq[T] {.magic: "ArrToSeq", noSideEffect.}
```
Which performs something else from `newSeq[byte](0)`, so if you do not like my legacy approach you could convert this line to
```nim
default(seq[byte])
```

Reply:
The implementation is here: 

https://github.com/nim-lang/Nim/blob/71ba2e7f3c5815d956b1ae0341b0743242b8fec6/compiler/ccgexprs.nim#L1576C29-L1576C29 

https://github.com/nim-lang/Nim/blob/71ba2e7f3c5815d956b1ae0341b0743242b8fec6/compiler/ccgexprs.nim#L1559

This follows from the compile-time knowledge of the length of an array - @ always uses their length to create sequences in the most efficient way possible (by pre-allocating a correctly lengthed seq or using `nil` when no allocation is needed).

We can test this theory in a simple way:

```nim
proc myfunc*() =
  var vvv: seq[byte] = @[]
  var vvv2: seq[byte] = default(seq[byte])
myfunc()```

```c
N_LIB_PRIVATE N_NIMCALL(void, myfunc__testit_3)(void) {
        tySequence__6H5Oh5UUvVCLiakt9aTwtUQ* vvv;
        tySequence__6H5Oh5UUvVCLiakt9aTwtUQ* vvv2;
        nimfr_("myfunc", "/home/arnetheduck/status/nimbus-eth2/_exper/testit.nim");
        vvv = NIM_NIL;
        vvv2 = NIM_NIL;
        popFrame();
}
```

Reply:
Ok, thank you, but let it will be styling requirement.

Reply:
60c3a42b7fedc492c785755161f73d407039df95